src/clientiface.cpp

```diff
@@ -185,6 +185,9 @@ void RemoteClient::GetNextBlocks (
 		m_blocks_occ.clear();
 	}
 	if (m_nearest_unsent_d > 0) {
+		if (!m_blocks_modified.empty()) {
+			m_blocks_occ.clear();
+		}
```

Comment:
```suggestion
		if (!m_blocks_modified.empty()) {
			m_blocks_occ.clear();
		}
```

Reply:
also is there any possibility that `m_nearest_unsent_d == 0` and `m_blocks_occ` wasn't already reset?
that'd cause bugs here

Reply:
This has been remove already.